lib/API/api_requests.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+Future<http.Response> getUserLists(String token) {
+  return http.post(
+    Uri.http('10.0.2.2:5000', '/get_lists'),
+    headers: <String, String>{
+      'Content-Type': 'application/json; charset=UTF-8',
+    },
+    body: jsonEncode(<String, String>{
+      'token': token,
+    }),
+  );
+}
+
+
+Future<http.Response> listAction(String token, String action, String listName) {
```

Comment:
Why is action a string? What kind of actions am I allowed to use? What happens if the action does not exist?

Reply:
the action in this function is a string to a specific endpoint. the user doesnt call this function, its only being called by the widgets in the app, so only actions that exist are being sent.

Reply:
I wasn't talking about users. I meant allowed to use as a developer working on extending this code. With this implementation, this is very inconvenient to use. I want you to explain why and then fix it.

Reply:
is it inconvenient because the extending developer doesn't know all of the endpoints?
the only way I can think of fixing it is by saving a list that contains all of the legal endpoints and at the beginning of the function check if the "action" exists in the array.